### PR TITLE
REV-024: create my roles page

### PR DESCRIPTION
**Pull request by RaniaSAloun**
Created my roles page with the following changes:

- added a new route for the page 
- created a page component in the pages folder
- created a table component (to be replaced with a shared component once built)
- created a cards container component, and a card component in the container component folder

Slide pages:
Desktop: https://optimumpartners.invisionapp.com/console/share/2Y34DAX6RD/819205008
Mobile: https://optimumpartners.invisionapp.com/console/share/3V3512T8G6/843334696

### CODE REVIEW

File: src/components/RoleCardsContainer/RoleCard/index.jsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        <Icon clicked={isClicked ? 'true' : undefined} />
+      </IconContainer>
+      <RoleText>{role}</RoleText>
+      <Overview>{overview}</Overview>
+    </Card>
+  );
+}
+
+RoleCard.propTypes = {
+  // eslint-disable-next-line react/forbid-prop-types
```
Comment (AseelRousan-op) [inline src/components/RoleCardsContainer/RoleCard/index.jsx:50]: can we git rid of this eslint disabling line here, 
```suggestion
 Icon: PropTypes.oneOf(['img', 'svg']).isRequired,
```

File: src/components/RolesTable/index.jsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import editIcon from 'assets/icons/edit.svg';
+import React, { useEffect, useState } from 'react';
+import getTableData from './data';
+import {
+  IconContainer,
+  StyledTable,
+  TableWrapper,
+  Td,
+  Th,
+  Tr,
```
Comment (AseelRousan-op) [inline src/components/RolesTable/index.jsx:10]: Can we Rename these components to have a meaningful name, for example:  TableCell, TableHead, TableRow

File: src/components/RoleCardsContainer/RoleCard/index.jsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import PropTypes from 'prop-types';
```
Comment (AseelRousan-op) [inline src/components/RoleCardsContainer/RoleCard/index.jsx:1]: Can we get the `RoleCard` component out of `RoleCardsContainer` directory and get these both components and the `RolesTable` component is `my-roles` directory so the structure will be something like this: 
my-roles=>RoleCardsContainer, RoleCard, RolesTable
and update the needed changes regarding this, please

**Review by AseelRousan-op — CHANGES_REQUESTED** [review]

File: src/components/my-roles/RolesTable/index.jsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      <StyledTable>
+        <thead>
+          <TableRow>
+            <TableHead>PROPERTY</TableHead>
+            <TableHead>MY ROLE</TableHead>
+            <TableHead />
+          </TableRow>
+        </thead>
+        <tbody>
+          {tableData.map((row) => (
```
Comment (mohdabed123) [inline src/components/my-roles/RolesTable/index.jsx:43]:
```suggestion
          {tableData.map(({ id, property, role }) => (
```

File: src/components/my-roles/RoleCardsContainer/role-cards-container.styles.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    flex-direction: row;
+    max-width: 52.5rem;
+  `};
+
+  ${mq.desktopExtraMax`
+    max-width: 78.75rem;
+  `};
+`;
+
+export const Icon = (agent) => styled(agent)`
```
Comment (mohdabed123) [inline src/components/my-roles/RoleCardsContainer/role-cards-container.styles.js:23]:
```suggestion
export const Icon = (icon) => styled(icon)`
```

File: src/components/my-roles/RoleCardsContainer/role-cards-container.styles.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+import colors from 'styles/colors';
+import mq from 'styles/media-query';
+
+export const CardsContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  flex-wrap: wrap;
+  justify-content: flex-start;
+  margin: 0 0 1.375rem;
+  width: 100%;
```
Comment (mohdabed123) [inline src/components/my-roles/RoleCardsContainer/role-cards-container.styles.js:11]: do we need this?

File: src/components/my-roles/RoleCardsContainer/role-cards-container.styles.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
+import styled from 'styled-components';
+import colors from 'styles/colors';
+import mq from 'styles/media-query';
+
+export const CardsContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  flex-wrap: wrap;
+  justify-content: flex-start;
```
Comment (mohdabed123) [inline src/components/my-roles/RoleCardsContainer/role-cards-container.styles.js:9]: is this needed?

File: src/components/my-roles/RoleCardsContainer/index.jsx [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    const fetchRoles = async () => {
+      const data = await getRoles();
+      setRoles(data);
+    };
+    fetchRoles();
+  });
+
+  return (
+    <CardsContainer>
+      {roles.map((role) => (
```
Comment (mohdabed123) [inline src/components/my-roles/RoleCardsContainer/index.jsx:31]:
```suggestion
      {roles.map(({ Icon, role, overview }) => (
```

File: src/components/my-roles/RoleCard/role-card.styles.js [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+export const Card = styled.div`
+  border-radius: 0.625rem;
+  border: 0.0625rem solid;
+  color: ${({ clicked }) =>
+    clicked ? `${colors.green}` : `${colors.dustyGrey}`};
+  border-color: ${({ clicked }) =>
+    clicked ? `${colors.green}` : `${colors.mercury}`};
+  cursor: pointer;
+  display: flex;
+  flex-direction: column;
```
Comment (mohdabed123) [inline src/components/my-roles/RoleCard/role-card.styles.js:15]: do we need these?

**Review by mohdabed123 — APPROVED** [review]
small changes other than that LGTM

File: src/components/my-roles/RolesTable/roles-table.style.js [hunk 10]
```diff
@@ -0,0 +1,45 @@
+import styled from 'styled-components';
+import colors from 'styles/colors';
+import mq from 'styles/media-query';
+
+export const TableWrapper = styled.div`
+  border-radius: 0.375rem;
```
[the remark this answers is not in the document]
Comment (RaniaSAloun) [reply src/components/my-roles/RolesTable/roles-table.style.js:6]: it adds a border to the table inside

File: src/components/my-roles/RoleCardsContainer/index.jsx [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+  });
+
+  return (
+    <CardsContainer>
+      {roles.map(({ Icon, role, overview }) => (
+        <RoleCard
+          Icon={Icon}
+          key={role}
+          overview={overview}
+          role={role}
```
Comment (AhmadAlShwaiki) [inline src/components/my-roles/RoleCardsContainer/index.jsx:36]: Please reduce the number of Props where possible
```suggestion
          key={role}
          data={ {role, overview, Icon} }
```

File: src/components/my-roles/RoleCard/role-card.styles.js [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+  background-color: ${({ clicked }) =>
+    clicked ? `${colors.green}` : `${colors.white}`};
+  border: ${({ clicked }) =>
+    clicked ? `none` : `0.125rem solid ${colors.dustyGrey}`};
+  color: ${colors.white};
+  height: 1.25rem;
+  position: absolute;
+  right: 1.25rem;
+  top: 0.9375rem;
+  width: 1.25rem;
```
Comment (AhmadAlShwaiki) [inline src/components/my-roles/RoleCard/role-card.styles.js:43]:
```suggestion
  border-radius: 0.625rem;
```

Comment (AhmadAlShwaiki) [inline src/components/my-roles/RoleCard/role-card.styles.js:43]:
```suggestion
  width: 1.25rem;
  border-radius: 0.625rem;
```

File: src/components/my-roles/RoleCard/index.jsx [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+        const next = new Set(prev);
+        next.delete(role);
+        return next;
+      });
+    setIsClicked(!isClicked);
+  };
+
+  return (
+    <Card onClick={handleCardSelect} clicked={isClicked}>
+      <Rectangle clicked={isClicked}>{isClicked && <CheckIcon />}</Rectangle>
```
Comment (AhmadAlShwaiki) [inline src/components/my-roles/RoleCard/index.jsx:42]:
```suggestion
      <Rectangle isSelected={isClicked}>{isClicked && <CheckIcon />}</Rectangle>
```

File: src/pages/MyRoles/my-roles.styles.js [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+  font-weight: 600;
+  line-height: 2.4375rem;
+`;
+
+export const StyledButton = styled(Button)`
+  font-size: 0.9375rem;
+
+  ${mq.desktop`
+    font-size: 1.125rem;
+  `}
```
Comment (AhmadAlShwaiki) [inline src/pages/MyRoles/my-roles.styles.js:38]:
```suggestion
  font-size: 1rem;
`;
```

File: src/components/my-roles/RoleCard/role-card.styles.js [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+`;
+
+export const CheckIcon = styled(checkIcon)`
+  height: 100%;
+  width: 100%;
+`;
+
+export const RoleText = styled.p`
+  font-weight: bold;
+`;
```
Comment (AhmadAlShwaiki) [inline src/components/my-roles/RoleCard/role-card.styles.js:58]: Can simply be an h3, no need for this component
```suggestion
export const RoleText = styled.h3`
`;
```

**Review by AhmadAlShwaiki — CHANGES_REQUESTED** [review]
some minor component and styling suggestions, otherwise looks really good.

**Comment by RaniaSAloun** [comment]
@AhmadAlShwaiki For the comment on the checkbox style, it's a design change request, was discussed with @AseelRousan-op, the cards are represented as check-boxes not radio buttons.

**Comment by RaniaSAloun** [comment]
@AhmadAlShwaiki 
for this comment: 'Can simply be an h3, no need for this component' on the RoleText styled component;
I've read on w3schools that we shouldn't use headings to make text big or bold, headings should be used for main headings to be used by search engines.
https://www.w3schools.com/html/html_headings.asp

Do you still want it replaced?

**Comment by AhmadAlShwaiki** [comment]
> @AhmadAlShwaiki for this comment: 'Can simply be an h3, no need for this component' on the RoleText styled component; I've read on w3schools that we shouldn't use headings to make text big or bold, headings should be used for main headings to be used by search engines. https://www.w3schools.com/html/html_headings.asp
> 
> Do you still want it replaced?

What matters here is that we shouldn't make/reinvent a styled component for each html element when we simply could use one. If you seek to match the advise mentioned on the website you can keep it.

**Review by AhmadAlShwaiki — APPROVED** [review]

**Review by mohdabed123 — APPROVED** [review]

File: src/components/my-roles/RoleCardsContainer/index.jsx [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+import PropTypes from 'prop-types';
+import React, { useEffect, useState } from 'react';
+import getRoles from './data';
+import { CardsContainer } from './role-cards-container.styles';
+
+/**
+ * Component holding role cards, it maps roles to their cards.
+ *
+ * @param {Object} props Props passed to RoleCardsContainer.
+ * @param {func} props.setSelectedRoles Sets selected roles.
```
Comment (AseelRousan-op) [inline src/components/my-roles/RoleCardsContainer/index.jsx:11]:
```suggestion
 * @param {Object} props                  Props passed to RoleCardsContainer.
 * @param {func}   props.setSelectedRoles Sets selected roles.
```

**Review by AseelRousan-op — APPROVED** [review]

**Review by AseelRousan-op — APPROVED** [review]
👍

**Review by Mahmoudzaid2994 — APPROVED** [review]

**Pull request merged**